clients/da-vinci-client/src/test/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTaskTest.java

```diff
@@ -2144,21 +2112,29 @@ public void testDataValidationCheckPointing(SortedInput sortedInput, AAConfig aa
 
     // This doesn't really need to be atomic, but it does need to be final, and int/Integer cannot be mutated.
     final AtomicInteger messagesConsumedSoFar = new AtomicInteger(0);
+    Runnable runnable = () -> {
+      relevantPartitions.stream()
+          .forEach(
+              partition -> storeIngestionTaskUnderTest
+                  .unSubscribePartition(new PubSubTopicPartitionImpl(pubSubTopic, partition))
+          );
+      relevantPartitions.stream()
+          .forEach(
+              partition -> storeIngestionTaskUnderTest
+                  .subscribePartition(new PubSubTopicPartitionImpl(pubSubTopic, partition), false)
+          );
+    };
     PollStrategy pollStrategy =
         new BlockingObserverPollStrategy(new RandomPollStrategy(false), topicPartitionOffset -> {
           if (topicPartitionOffset == null || topicPartitionOffset.getOffset() == null) {
             LOGGER.info("Received null OffsetRecord!");
           } else if (messagesConsumedSoFar.incrementAndGet()
               % (totalNumberOfMessages / totalNumberOfConsumptionRestarts) == 0) {
             LOGGER.info("Restarting consumer after consuming {} messages so far.", messagesConsumedSoFar.get());
-            relevantPartitions.stream()
-                .forEach(
-                    partition -> storeIngestionTaskUnderTest
-                        .unSubscribePartition(new PubSubTopicPartitionImpl(pubSubTopic, partition)));
-            relevantPartitions.stream()
-                .forEach(
-                    partition -> storeIngestionTaskUnderTest
-                        .subscribePartition(new PubSubTopicPartitionImpl(pubSubTopic, partition)));
+
+            Thread infiniteLockThread =
```

Comment:
This is to simulate the async process, right? But why name it as infiniteLockThread?